Lib/ufo2ft/kernFeatureWriter.py

```diff
@@ -109,80 +92,54 @@ def _collectFeaClassKerning(self):
             for rightName, rightContents in self.rightFeaClasses.items():
                 rightKey = rightContents[0]
                 pair = leftKey, rightKey
-                kerningVal = self.kerning[pair]
+                kerningVal = self.kerning.get(pair)
                 if kerningVal is None:
                     continue
                 self.classPairKerning[leftName, rightName] = kerningVal
-                self.kerning.remove(pair)
+                del self.kerning[pair]
 
             # collect rules with left class and right glyph
-            for pair, kerningVal in _getGlyphKern(self.kerning, leftKey, 0):
+            for pair, kerningVal in self._getGlyphKerning(leftKey, 0):
                 self.leftClassKerning[leftName, pair[1]] = kerningVal
-                self.kerning.remove(pair)
+                del self.kerning[pair]
 
         # collect rules with left glyph and right class
         for rightName, rightContents in self.rightFeaClasses.items():
             rightKey = rightContents[0]
-            for pair, kerningVal in _getGlyphKern(self.kerning, rightKey, 1):
+            for pair, kerningVal in self._getGlyphKerning(rightKey, 1):
                 self.rightClassKerning[pair[0], rightName] = kerningVal
-                self.kerning.remove(pair)
-
-    def _collectUfoClasses(self):
-        """Sort UFO groups into left or right glyph classes."""
-
-        for name, contents in self.groups.items():
-            if self._isClassName(self.leftUfoGroupRe, name):
-                self.leftUfoClasses[name] = contents
-            if self._isClassName(self.rightUfoGroupRe, name):
-                self.rightUfoClasses[name] = contents
+                del self.kerning[pair]
 
     def _correctUfoClassNames(self):
         """Detect and replace OTF-illegal class names found in UFO kerning."""
```

Comment:
It's really FEA-illegal and not OTF here. Classes aren't named inside an OTF (ClassDef) so there would be no naming restriction if it wasn't for the use of FEA as an intermediary format.